### ISSUE TITLE
Draining the event queue on Canon 700D

### ISSUE THREAD

**Issue by mihailtudoran** [gphoto/libgphoto2#346]
Hi,

Canon 700d seems to flood with events even when idle, after 12 hours the event queue ends up being about 8000 events ( +/- 500 ). When taking a new photo, first it is necessary to drain the old events from the camera, operation that takes about 10 seconds.

Is there a way to ignore non FILE_ADDED events or drain the queue faster ? Currently I am polling for all events prior of taking the picture in order to drain the queue. 

Thanks!

**Comment by msmeissn** [comment]
WIth 8000 events the drain algorithm shows its quadratic O(n^2) nature sadly.. This might be fixable.

can you drain it inbetween the waiting?

**Comment by mihailtudoran** [comment]
Yes, a workaround would be to have a thread that constantly drains the event queue and stopping it right before the actual capture takes place.

Need to mention this is not happening on Canon 600D as the camera don't issue new event when idle. I believe the 700D is sending a periodic heartbeat or something I cannot get rid of .

**Comment by mihailtudoran** [comment]
Most effective workaround is to simply reconnect the camera. It will drain all events.
